mozilla-release/toolkit/modules/CliqzLogger.jsm

```diff
@@ -0,0 +1,48 @@
+/* global ChromeUtils */
+"use strict";
+
+var { Services } = ChromeUtils.import("resource://gre/modules/Services.jsm");
+
+const EXPORTED_SYMBOLS = ["CliqzLogger"];
+
+const CliqzLogger = (function() {
+  // 0 - do not log any information;
+  // 1 - display it on a screen;
+  // 2 - write down to the file;
+  const PREF_NAME = "app.log.enabled.level";
+
+  let level = 0;
+  let prefChangeObserver = {
+    observe: function(subject, topic, prefName) {
+      level = Services.prefs.getIntPref(prefName, 0);
+    }
+  };
+
+  Services.prefs.addObserver(PREF_NAME, prefChangeObserver);
+
+  if (Services.prefs.prefHasUserValue(PREF_NAME)) {
+    level = Services.prefs.getIntPref(PREF_NAME);
+  } else {
+    Services.prefs.setIntPref(PREF_NAME, level);
+  }
+
+  return {
+    init: function(modulePath, moduleName) {
+      return function CliqzSaveLog() {
+        if (level === 0) {
+          return;
+        } else if (level === 1) {
+          let messages = Array.prototype.slice.call(arguments);
+
+          messages.unshift(modulePath);
+          console.log.apply(console, messages);
+        }
+      };
+    },
+    destroy: function() {
+      Services.prefs.removeObserver(PREF_NAME, prefChangeObserver);
+      level = null;
```

Comment:
Maybe this should be `0`?

Reply:
@remusao I am not sure that Garbage Collector will come after 0 values.